### PR TITLE
Add github checks api data models.

### PR DESCRIPTION
**Pull request by godofredoc**
Bug:
  https://github.com/flutter/flutter/issues/56422

### CODE REVIEW

File: app_dart/test/model/github/test_data/check_suite.json [hunk 1]
```diff
@@ -0,0 +1,222 @@
+{
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/test/model/github/test_data/check_suite.json:1]: Actually I need to remove those files. They can not be loaded from the file directly and need to get embedded in the dart file.

File: app_dart/test/model/github/checks_test.dart [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      expect(pullRequest.base.repo.name, equals('cocoon'));
+      expect(pullRequest.head.ref, equals('update_licenses'));
+      expect(pullRequest.head.sha,
+          equals('5763f4c2b3b5e529f4b35c655761a7e818eced2e'));
+      expect(pullRequest.head.repo.name, equals('cocoon'));
+    });
+  });
+}
+
+const String checkSuiteString = '''\
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/test/model/github/checks_test.dart:37]: I tried but I couldn't find a way to load resource files directly from dart. Tests are encoded before running and there is not way to make a reference to the local files.

[the remark this answers is not in the document]
Comment (keyonghan) [reply app_dart/test/model/github/checks_test.dart:37]: alternatively, you can try saving this json variable in a separate dart file. Similar to: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/check_for_waiting_pull_requests_queries.dart
Then import that file in the test.

File: app_dart/lib/src/model/github/checks.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:github/github.dart' hide CheckSuite;
+import 'package:github/hooks.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'checks.g.dart';
+
+@JsonSerializable(fieldRename: FieldRename.snake)
```
Comment (digiter) [inline app_dart/lib/src/model/github/checks.dart:11]: Nit: provide links to github checks api in comments.

Comment (godofredoc) [reply app_dart/lib/src/model/github/checks.dart:11]: done

File: app_dart/test/model/github/checks_test.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/test/model/github/checks_test.dart:37]: That's a great idea, I'll move them to  a different file in the next 
PR adding checkruns models.

**Pull request merged**